frontend/src/app/components/crm/leads/leads.component.ts

```diff
@@ -206,17 +207,12 @@ export class LeadsComponent implements AfterViewInit {
     return this.selection.selected.length > 0;
   }
 
-  openDialog() {
-      const dialogRef = this.dialog.open(DetailsDialog);
-
-      dialogRef.afterClosed().subscribe(result => {
+  openDialog(leadIdCheck) {
+   const toBeDisplayed = this.dataSource.data.filter(toOpen => toOpen.leadId == leadIdCheck);
+   let dialogRef = this.dialog.open(LeadDetailsComponent, {
+        width: '750px',
+        data: { details: toBeDisplayed }
```

Comment:
Pass the lead object itself rather than the id to the `openDialog` function. Then you won't have to filter for it.

Reply:
Alright and that will also make the code shorter